### PR TITLE
Added blob_client for uploading blob

### PR DESCRIPTION
**Pull request by thezultimate**
`SumoClient` contains a `put` method for uploading blobs via `sumo-core`. To allow uploading blobs directly to Azure, a new `BlobClient` class is added in this PR.

### CODE REVIEW

File: src/sumo/wrapper/_blob_client.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import requests
+
+class BlobClient:
+    def __init__(self, raise_request_error_function):
+        self._raise_request_error_exception = raise_request_error_function
+        pass
```
Comment (adnejacobsen) [inline src/sumo/wrapper/_blob_client.py:6]: We can remove `pass`?

**Review by adnejacobsen — APPROVED** [review]

**Review by adnejacobsen — CHANGES_REQUESTED** [review]
Move `_raise_request_error_exception` into `_request_error.py` and import into `SumoClient` and `BlobClient`

**Comment by thezultimate** [comment]
> 

This has been applied, please check again :)

**Review by adnejacobsen — APPROVED** [review]
LGTM!

**Pull request merged**